### DISCUSSION TITLE
Is there a way to customize the default filter?

### DISCUSSION THREAD

**Discussion by michaelschwob** [joonaspaakko/audible-library-extractor#80] [Q&A] [answered]
As the title suggests, I think it would be cool if there was a way to adjust the default filter.

Thank you for building such a wonderful tool! :)

**Comment by joonaspaakko** [comment]
I suppose it depends what you mean by filter customization. The filter menu is the second icon on the right side of the search bar. So I guess that is where you would change the filtering. Though if you feel it's missing something, I'm open to suggestions.

> It does have a limitation where all selected filters must apply to the book. For example if you pick `Whispersync: owned` + `Whispersync: available`, it wouldn't show you anything because the value can't be both at the same time.

![20E49977-CBEC-4A13-A2E9-31A4C9FB18A8](https://user-images.githubusercontent.com/1164476/152626861-9e8f5d5e-9257-4050-bf01-8a9554cad3cc.png)

**Reply by michaelschwob** [reply]
What I mean is that it would be nice to set which filters we want to be applied as the default so whenever someone goes to the page, (without filtering things themselves) they will see the filters that I have set as the default.

Regardless, this is one of the coolest tools I have found on GitHub so far, so I love it!

**Reply by joonaspaakko** [reply]
Glad you like it :)

**Comment by joonaspaakko** [comment, marked as answer]
Every time you change certain key things (listed below), the changes are saved in the URL as parameters.

-  `search scope`
- `filters`
- `sorting`
- `open/close book details`
- `use the search`
- _(and a few more)_

### What does it mean?
Basically whenever you reload the page or return back to the exact URL, it continues where you left off and because of this you can sort of force a default view/filtering by first opening the page, filtering stuff and then either sharing a link to the standalone gallery uploaded online or for your own use, **bookmark the page**. On mobile you can use regular bookmarks too, but the "Add to home screen" feature in all native mobile browsers is even better.

In the extension gallery right now the only way to set a default langing page is to make a bookmark and open it using that. The upcoming version `v.0.2.9` will have a functionality to set the default landing page, which follows the same logic as explained above, except that this controls the page that is opened via any links to the gallery provided by the extension, like the [extension/browser context menu item](https://joonaspaakko.gitbook.io/audible-library-extractor/gallery/opening-the-gallery).

This is the current draft of the feature in `v.0.2.9`
 ![extension-gallery-change-default-langing-page](https://user-images.githubusercontent.com/1164476/185054228-fa70ca62-322b-44ef-b50d-9111244311b2.png)
